Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepContext.cs

```diff
@@ -40,34 +40,27 @@ public sealed class BlockStoreInnerStepContext
         /// <summary>Timestamp of the last flush of <see cref="DownloadStack"/> to the disk.</summary>
         internal DateTime LastDownloadStackFlushTime;
 
-        public BlockStoreInnerStepContext(CancellationToken cancellationToken, BlockStoreLoop blockStoreLoop, ILoggerFactory loggerFactory, IDateTimeProvider dateTimeProvider)
+        public BlockStoreInnerStepContext(CancellationToken cancellationToken, BlockStoreLoop blockStoreLoop, ChainedBlock nextChainedBlock, ILoggerFactory loggerFactory, IDateTimeProvider dateTimeProvider)
         {
             Guard.NotNull(blockStoreLoop, nameof(blockStoreLoop));
+            Guard.NotNull(nextChainedBlock, nameof(nextChainedBlock));
 
             this.loggerFactory = loggerFactory;
             this.logger = loggerFactory.CreateLogger(GetType().FullName);
-            this.DateTimeProvider = dateTimeProvider;
+            this.logger.LogTrace("({0}:'{1}/{2}')", nameof(nextChainedBlock), nextChainedBlock?.HashBlock, nextChainedBlock?.Height);
```

Comment:
We do have a guarded guarantee of nextChainedBlock not being `null` here. So you can remove question marks.